### ISSUE TITLE
ReactorCraft Boilers in hell can conduct heat to Boilers that are not in hell biomes

### ISSUE THREAD

**Issue by Prunoideae** [ReikaKalseki/Reika_Mods_Issues#1998]
It looks like a really absurd issue but actually it can cause infinite steam output when have both REC and CC installed...
I can convert a small area in overworld to hell biome and place some boilers on the 'border' like this
![tim 20181122154301](https://user-images.githubusercontent.com/24620047/48888334-546bd880-ee6d-11e8-8c01-875a479bbbdc.jpg)
and when I placed a boiler outside this area, the boiler in hell would transfer heat to the boiler in the normal biome, then I can make steam by this boiler because it will never explode...
![tim 20181122154343](https://user-images.githubusercontent.com/24620047/48888442-ac0a4400-ee6d-11e8-912b-34f1e88aee9e.jpg)
It's a little bit slow to wait for the boiler in hell to accumulate heat, but if used with a block breaker and a block placer, and some sort of redstone circuit, the boiler's temperature can be locked at almost 300, which cause a really efficient steam output.
Making steam in this way should not be encouraged because it almost consume nothing, everything you need just some HSLA steel and tungsten carbide, you don't EVEN need Cd-In-Ag alloy because there are no Fuel Cores in this structure.

**Comment by Prunoideae** [comment]
It does output steam
![tim 20181122155123](https://user-images.githubusercontent.com/24620047/48888704-82055180-ee6e-11e8-8945-0e101deadd1c.jpg)

**Comment by ReikaKalseki** [comment]
I can probably fix this by making the nether check also forbid the overworld. A more intensive solution is for reactors to cache their overall biome, but that is...potentially risky.

**Comment by Prunoideae** [comment]
Also, is it 'legal' to use Sodium Heater in hell and transfer heated sodium to overworld by teleporters like Dimensional Transreceiver or Tesseract? This can also reduce the temperature requirement a lot.
And I can also paint the sodium heater...

**Comment by ReikaKalseki** [comment]
No, that is also cheating. But this is rapidly approaching the point where fixability is uncertain.

**Comment by MachiavelliSeraphim** [comment]
Wow.  
Thats sort of sad because I thought it was genuinely inventive.
Ive been using a hybrid breeder/fission core setup in hell to produce hot sodium snd plutonium for years.  
Then tesseract/transport the sodium to remote locations for distributed generation.

**Comment by ReikaKalseki** [comment]
How is it news that using a boiler and leveraging temperature mechanics without actually having a "working" reactor is cheating?

**Comment by OvermindDL1** [comment]
> How is it news that using a boiler and leveraging temperature mechanics without actually having a "working" reactor is cheating?

Ehh, honestly I would think it's by design too.

However, I'm not sure how the Nether's/hell temperature is that hot, standard igneous(sp?) rocks only melt into lava at some X degrees, which is significantly cooler than most metal's melting temperatures, so Hell's temperature shouldn't actually be hot enough to melt most things.  However, any kind of thermolitic reaction can generate power from temperature differentials, so if you have a perfect temperature differential then you have unlimited power.  What would be interesting is that while the 'temperature' of a biome is drained then the biome should be changed slowly to equalize the temperatures, so hotter ones get colder 'at' the location of whatever is draining the heat, and colder biomes get warmer.  That would be both a balancing measure, get rid of the infinity of the generation, and could be an interesting way to very slowly re-biome areas.

**Comment by ReikaKalseki** [comment]
> Ehh, honestly I would think it's by design too.

*blank stare*

Why is it not assumed that it is intended that you need a nuclear reaction for your nuclear reactor to make power?

> However, I'm not sure how the Nether's/hell temperature is that hot, standard igneous(sp?) rocks only melt into lava at some X degrees, which is significantly cooler than most metal's melting temperatures, so Hell's temperature shouldn't actually be hot enough to melt most things.

What does melting metal have to do with anything?

> What would be interesting is that while the 'temperature' of a biome is drained then the biome should be changed slowly to equalize the temperatures, so hotter ones get colder 'at' the location of whatever is draining the heat, and colder biomes get warmer.

Putting aside that I doubt such a thing can even be detected (how can a reactor block in the nether know that it is connected, via mod pipe X and interdimensional link Y, to the overworld?), you are suggesting effectively 'freezing' the nether. That does not make sense given that it is a near-infinite reservoir of its temperature.

**Comment by OvermindDL1** [comment]
> Why is it not assumed that it is intended that you need a nuclear reaction for your nuclear reactor to make power?

Well in real life a nuclear reactor is made up of a few parts, it's not a single magical block that some mods like to do \*cough\*IC2\*cough\*.  Basically it's just a set of properly made uranium setup to maximize neutron reactions with control/absorption rods between to mediate the flow and the neutron's just react to the absorption surface on the outside to generate heat (with this entire assembly submerged in a coolant to keep the reaction rods cool), this whole setup so far can be tiny and is only the first part.  The heat that it generates on the outer absorption surface is then used to generate power, either via thermolytic reactions or by something as simple as (and this is the most common) heating water to steam and using the steam to spin turbines.

Thus a nuclear reactor is just something to make Heat, that's it; the Heat then being used to generate power any other way you'd make heat (such as by burning coal in a furnace or something), such as boiling water.  It doesn't matter how you get heat, such as via some form of hell dimension, if you have some way to send it to some place cool (like melting salt and taking that melted salt to a cold environment) then you can use it to do work, such as heating water (or just heat the water directly in whatever 'hell' dimension assuming such a dimension existed in real life).  The process of making 'work' (electricity, spinning things, whatever) is just a simple transference from a high heat to low heat area, that transference of the heat itself is what does the work (after that the two areas should be in temperature equilibrium excepting external influence like hot/cold air adjusting it further).

In real life if you could get such a 'hell' dimension that has a natural high heat then you could just ship water to it to convert to steam then do work with that steam, essentially you are bringing 'cold' to the hell dimension, and you made a temperature gradiant between the cold water and the hot environment and thus work is performed, which in water's case turns to steam, which you can then use to do further work (based on the amount of joules of energy difference between the temperatures and the expansion of the water, which is quite substantial, water->steam conversions are pretty crazy efficient hence why almost all power generated nowadays is via a water->steam conversions then sending the steam to turbines before releasing or condensing it back, you couldn't really do the condensing stage in a 'hell' dimension though so you'd either have to ship it back and keep shipping water in or ship even more water in without the closed loop and just release the steam after use).

So yes, being able to take 'heat' from a hell dimension back to something our normal temperatures I would entirely expect to be able to do actual 'work'.  :-)

> What does melting metal have to do with anything?

Mostly just arguing at how some mods \*cough\*IC2\*cough\*TE\*cough\* pretend that surface lava is hot enough to melt something like iron, which is just stupid...

> Putting aside that I doubt such a thing can even be detected (how can a reactor block in the nether know that it is connected, via mod pipe X and interdimensional link Y, to the overworld?),

You wouldn't, anything that 'takes' heat and packages it up into something (like melted salts) would only be able to do that work by cooling the space it is in anyway (well, bringing to equilibrium the environment and the salt), which you could 'emulate/fake' in the nether by changing the biome on that one column.

> you are suggesting effectively 'freezing' the nether.

Eyup, in a column by column basis.

> That does not make sense given that it is a near-infinite reservoir of its temperature.

I entirely agree, in the real world that makes sense, but then again this is Minecraft that can have a single block of falling snow amidst a jungle and those don't "balance out", so it makes conceptual sense given Minecraft's existing funkiness.

As an aside, if you remember RedPower2, Eloraam has a thermopile generator that took heat on one side and cold on the other to generate power (using up the temperature gradiant over time, so ice would turn to water, lava would solidify, etc...), if you found a cold and hot biome next to each other then you could make infinite (though little amounts) of power pretty easily in RP2 (the thermopile was massively nerfed so you could never generate large amounts of power with it, so it was only sufficient for things like running Sorters or Computers or so even with a nether biome on one side and an ice biome on the other of a thermopile).

**Comment by ReikaKalseki** [comment]
> So yes, being able to take 'heat' from a hell dimension back to something our normal temperatures I would entirely expect to be able to do actual 'work'. :-)

You explained why from a mechanical standpoint. But from a game design standpoint, does that sound like the kind of thing one would choose to add? A basically free massive power source? One that is so strong and so cheap that it supplants literally everything else, as soon as you have Nether access, forever?

> You wouldn't, anything that 'takes' heat and packages it up into something (like melted salts) would only be able to do that work by cooling the space it is in anyway (well, bringing to equilibrium the environment and the salt), which you could 'emulate/fake' in the nether by changing the biome on that one column.

Cooling the entire column has nothing to do with using its own internal heat.

> I entirely agree, in the real world that makes sense, but then again this is Minecraft that can have a single block of falling snow amidst a jungle and those don't "balance out", so it makes conceptual sense given Minecraft's existing funkiness.

I actually do implement such balancing, in ReC and elsewhere. Not for vanilla blocks, but for my own. So "it's MC, ignore the laws of physics" does not apply.

**Comment by MachiavelliSeraphim** [comment]
I have never used the nether to "cheat", i.e. get something for nothing.

I actually spent several months balancing my reactor designs so that it would be stable in the locations I was going to run them.

They all still burn fuel and produce waste.
The only oddity I put to use was the higher ambient temperature of hell and all it does in my case is reduce fuel consumption.

If you were to use chromaticraft instead of AE2 or TE, world rifts and large storage tanks could replace the infrastructure and redstone controls, so yes, I also assumed that the mechanics of running a reactor in hell and creating an extradimensinal sodium loop were intended.

Fusion reactors, once operational and self sustaining, create essentially free energy.

In summary, yes what is shown here would be cheating as I see it, but running extradimensional sodium loops is just being efficient at the cost of complexity

**Comment by Prunoideae** [comment]
No, it's still cheating because sodium heater outputs hot sodium at the temperature of 300, which is also the environmental temperature of hell, so I can use only two cores in the reactor and heat the reactor mainly by the ambient temperature of the hell. It could cut the need sharply.
Though I think that there's nothing but pebblebed-breeder-plutonium-fusion chain if this issue was fixed...

**Comment by MachiavelliSeraphim** [comment]
Try running a turbine off of something that small.  
It Wont happen.

Were getting to the point of arguing semantics.

Should a boiler/heater be able to make steam/sodium with no nuclear heat input?
NO.

Is it feasible to build the cheaty setup shown in the bug report?
Maybe/Probably.

Is it ludicrous to use hell as an efficiency boost?
I dont think so at all, its only a 260 degree booster, not a miracle.

I dont consider a reactor with 36 cores and 60 sodium heaters "small" or "simple".
Overworld it can drive a 2 3/4 or so HPT.
Nether it can drive 3 1/2 or so HPT.

I also have 3 dedicated AE2 systems to process fuel, store and direct sodium, and control core fueling off of my sodium and fuel levels in addition to the slew of TE ducts and Tesseracts to transport everything.

I spent months designing, tweaking, and engineering the plant I have.  I didnt take any shortcuts nor did I ever show anyone else how to cookie cutter this setup.

So I would argue that the engineering and cost of building a reactor complex like mine warrants the increase in efficiency of 3/4 of a turbine.

So from my point of view, its just an efficiency increase.  
I dont cheat the system, I just find the best way to achieve what I want using whats available.  

But in the end, yes, there should be a way to prevent the creation of something from nothing.

**Comment by OvermindDL1** [comment]
> Is it ludicrous to use hell as an efficiency boost?
> I dont think so at all, its only a 260 degree booster, not a miracle.

Technically you are just using the nether like a geothermal energy source, which are super valuable 'in the real world' too.  ;-)

**Comment by MachiavelliSeraphim** [comment]
After much consideration....
I think the simplest way to solve all of your concerns and the issue listed is to prevent the placement of reactor core components in hell biomes.  
If we are not supposed to transport out sodium or CO2 to the overworld, then there isnt a reason I can see to allow the placement of core components in hell biomes.

**Comment by ReikaKalseki** [comment]
Tentatively fixed.

**Comment by MachiavelliSeraphim** [comment]
May I ask what your final implemented  choice to fix this problem is?
Just curious.

Thanks.

**Comment by ReikaKalseki** [comment]
Hard-clamping the ambient temperature to 95C or less when not in dimension -1.
